src/ucp/api/ucp.h

```diff
@@ -1151,14 +1155,20 @@ ucs_status_ptr_t ucp_tag_recv_nb(ucp_worker_h worker, void *buffer, size_t count
  *                          "ucp_tag_msg_recv_nb()" in order to receive the data
  *                          and release the resources associated with the
  *                          message handle.
+ *                          If false (0), the return value is merely an indication
+ *                          to whether a matching message is present, and it cannot
+ *                          be used in any other way, and in particular it cannot
+ *                          be passed to @ref ucp_tag_msg_recv_nb().
  * @param [out] info        If the matching message is found the descriptor is
  *                          filled with the details about the message.
  *
  * @return NULL                      - No match found.
  * @return Message handle (not NULL) - If message is matched the message handle
- *                                   is returned.
+ *                                     is returned.
  *
- * @todo Clarify release logic for remote == 0.
+ * @note This function does not make progress on the network. If this function
```

Comment:
I would proposed to use similar language. Progress = advance communication state.